src/Stream/src/BinderRabbitMQ/RabbitMessageChannelBinder.cs

```diff
@@ -137,10 +143,13 @@ protected override IMessageHandler CreateProducerMessageHandler(IProducerDestina
             var prefix = extendedProperties.Prefix;
             var exchangeName = destination.Name;
             var destinationName = string.IsNullOrEmpty(prefix) ? exchangeName : exchangeName[prefix.Length..];
-            var endpoint = new RabbitOutboundEndpoint(ApplicationContext, BuildRabbitTemplate(extendedProperties, errorChannel != null), _logger)
-            {
-                ExchangeName = exchangeName
-            };
+            var endpoint = new RabbitOutboundEndpoint(
+                ApplicationContext,
+                BuildRabbitTemplate(extendedProperties, errorChannel != null || extendedProperties.UseConfirmHeader.GetValueOrDefault()),
+                _logger)
```

Comment:
This line is getting complicated, would something like this make it easier to read?
```suggestion
            var template = BuildRabbitTemplate(extendedProperties, errorChannel != null || extendedProperties.UseConfirmHeader.GetValueOrDefault());
            var endpoint = new RabbitOutboundEndpoint(ApplicationContext, template, _logger)
```